src/main/java/frc/robot/subsystems/drivetrain/Drivetrain.java

```diff
@@ -1,10 +1,155 @@
-package frc.robot.subsystems.drivetrain;
-
-import com.ctre.phoenix.motorcontrol.can.TalonFX;
-
-public class Drivetrain {
-    private final TalonFX drivetrainMaster1=new TalonFX(1);
-    private final TalonFX drivetrainMaster2=new TalonFX(2);
-    private final TalonFX drivetrainMaster3=new TalonFX(3);
-    private final TalonFX drivetrainMaster4=new TalonFX(4);
-}
+package frc.robot.subsystems.drivetrain;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.can.TalonFX;
+import edu.wpi.first.wpilibj.Solenoid;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.Ports;
+import frc.robot.Robot;
+import frc.robot.subsystems.UnitModel;
+
+/**
+ * Build 4 motors, 2 for each side.
+ */
+public class Drivetrain extends SubsystemBase {
+    private final TalonFX masterRight = new TalonFX(Ports.Drivetrain.MASTER_RIGHT);
+    private final TalonFX slaveRight = new TalonFX(Ports.Drivetrain.SLAVE_RIGHT);
+    private final TalonFX masterLeft = new TalonFX(Ports.Drivetrain.MASTER_LEFT);
+    private final TalonFX slaveLeft = new TalonFX(Ports.Drivetrain.SLAVE_LEFT);
+    public UnitModel highGear = new UnitModel(Constants.Drivetrain.TICKS_PER_METER_HIGH_GEAR);
+    private UnitModel lowGear = new UnitModel(Constants.Drivetrain.TICKS_PER_METER_LOW_GEAR);
+    private Solenoid piston = new Solenoid(Ports.Drivetrain.SOLENOID);
+
+    /**
+     * Invert masterRight & slaveRight motors and set both slaves to follow and set sensor phase.
+     */
+    public Drivetrain() {
+        masterRight.setInverted(Constants.Drivetrain.MASTER_RIGHT_INVERTED);
+        masterRight.setSensorPhase(Constants.Drivetrain.MASTER_RIGHT_SENSOR_PHASE);
+        slaveRight.follow(masterRight);
+        slaveRight.setInverted(Constants.Drivetrain.SLAVE_RIGHT_INVERTED);
+        slaveRight.setSensorPhase(Constants.Drivetrain.SLAVE_LEFT_SENSOR_PHASE);

```

Comment:
The constant is `SLAVE_LEFT_SENSOR_PHASE` instead of `SLAVE_RIGHT_SENSOR_PHASE` and also, the slaves don't need to use `set sensor phase` method